test/e2e/autotls/auto_tls_test.go

```diff
@@ -231,18 +318,174 @@ func turnOffAutoTLS(t *testing.T, tlsClients *autoTLSClients) {
-	if err := wait.Poll(10*time.Second, 300*time.Second, func() (bool, error) {
+func waitForHTTP01ChallengePath(t *testing.T, tlsClients *autoTLSClients, objects *v1test.ResourceObjects) {
+	certName := routenames.Certificate(objects.Route)
+	if err := wait.Poll(5*time.Second, 60*time.Second, func() (bool, error) {
 		cert, err := tlsClients.clients.NetworkingClient.Certificates.Get(certName, metav1.GetOptions{})
+		cert, ready, err := getCertificate(t, tlsClients, certName)
+		if !ready {
+			return ready, err
+		}
+		if len(cert.Status.HTTP01Challenges) == 0 {
+			return false, nil
+		}
+		ingress, err := tlsClients.clients.NetworkingClient.Ingresses.Get(objects.Route.Name, metav1.GetOptions{})
 		if err != nil {
-			if apierrors.IsNotFound(err) {
-				t.Logf("Certificate %s has not been created: %v", certName, err)
-				return false, nil
-			}
 			return false, err
 		}
+		for _, rule := range ingress.Spec.Rules {
+			for _, path := range rule.HTTP.Paths {
+				if path.Path == cert.Status.HTTP01Challenges[0].URL.Path {
+					return true, nil
+				}
+			}
+		}
+		return false, nil
+	}); err != nil {
+		t.Fatalf("HTTP01 challenge path is not populated in Ingress: %v", err)
+	}
+}
+
+func waitForCertificateReady(t *testing.T, tlsClients *autoTLSClients, certName string) {
+	if err := wait.Poll(10*time.Second, 300*time.Second, func() (bool, error) {
+		cert, ready, err := getCertificate(t, tlsClients, certName)
+		if !ready {
+			return ready, err
+		}
 		return cert.Status.IsReady(), nil
 	}); err != nil {
 		t.Fatalf("Certificate %s is not ready: %v", certName, err)
 	}
 }
+
+func getCertificate(t *testing.T, tlsClients *autoTLSClients, certName string) (*v1alpha1.Certificate, bool, error) {
+	cert, err := tlsClients.clients.NetworkingClient.Certificates.Get(certName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			t.Logf("Certificate %s has not been created: %v", certName, err)
+			return nil, false, nil
+		}
+		return nil, false, err
+	}
+	return cert, true, nil
+}
+
+func setupDNSRecord(t *testing.T, cfg config, tlsClients *autoTLSClients, domain string) context.CancelFunc {
+	ip, err := ingress.GetIngressEndpoint(tlsClients.clients.KubeClient.Kube)
+	if err != nil {
+		t.Fatalf("Failed to get Gateway IP address %v.", err)
+	}
+	dnsRecord := &dnsRecord{
+		domain: domain,
+		ip:     ip,
+	}
+	if err := createDNSRecord(cfg, dnsRecord); err != nil {
+		t.Fatalf("Failed to create DNS record: %v", err)
+	}
+	if err := waitForDNSRecordVisibleLocally(dnsRecord); err != nil {
+		deleteDNSRecord(t, cfg, dnsRecord)
+		t.Fatalf("Failed to wait for DNS record to be visible: %v", err)
+	}
+	t.Logf("DNS record %v was set up.", dnsRecord)
+	return func() {
+		deleteDNSRecord(t, cfg, dnsRecord)
+	}
+}
+
+func waitForDNSRecordVisibleLocally(record *dnsRecord) error {
+	return wait.PollImmediate(10*time.Second, dnsRecordDeadline*time.Second, func() (bool, error) {
+		ips, err := net.LookupHost(record.domain)
+		if err != nil {
```

Comment:
Since you're ignoring the error, kind of, then just
```
if ips, _ := net.LookupHost(...)
for _, ip := range {
...
}
```

Reply:
Done